version: "2.1"

services:
  catalog_db:
    image: mysql:8.0.33
    environment:
      MYSQL_ROOT_PASSWORD: "pass"
      MYSQL_DATABASE: "ScoringSystem"
      MYSQL_USER: "user"
      MYSQL_PASSWORD: "password"
    ports:
      - "3306:3306"
    volumes:
      - "catalog_data:/var/lib/mysql"
    networks:
      - rpiserver
    healthcheck:
        test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
        timeout: 10s
        retries: 10

  redis_server:
    image: redis:alpine
    environment:
      REDIS_PASSWORD: pass
    ports:
      - "6379:6379"
    volumes:
      - "redis_data:/data"
    networks:
      - rpiserver

  api:
    image: localhost:5000/scoring-system-api:latest
    environment:
      DB_URL: "mysql+mysqlconnector://root:pass@catalog_db:3306/ScoringSystem"
      REDIS_URL: "redis://redis_server:6379"
      C_FORCE_ROOT: "true"
    depends_on:
      catalog_db:
        condition: service_healthy
    ports:
      - "8080:8080"
    networks:
      - rpiserver

  # celery worker with ml models simulation
  worker:
    image: localhost:5000/scoring-system-worker:latest
    environment:
      REDIS_URL: "redis://redis_server:6379"
      C_FORCE_ROOT: "true"
    networks:
      - rpiserver

  frontend:
    image: localhost:5000/scoring-system-frontend:latest
    ports:
      - "3001:3001"
    networks:
      - rpiserver

  proxy:
    image: nginx:alpine 
    ports:
      - "9123:9123"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    networks:
      - rpiserver
    depends_on:
      - api
      - frontend

volumes:
  catalog_data:
  nginx.conf:
  redis_data:

networks:
  rpiserver:
    external: true

